builder/azure/arm/config.go

```diff
@@ -774,6 +781,23 @@ func assertTagProperties(c *Config, errs *packer.MultiError) {
 func assertRequiredParametersSet(c *Config, errs *packer.MultiError) {
 	c.ClientConfig.Validate(errs)
 
+	/////////////////////////////////////////////
+	// Identity
+	if len(c.UserAssignedManagedIdentities) != 0 {
+		for _, rid := range c.UserAssignedManagedIdentities {
+			r, err := client.ParseResourceID(rid)
+			if err != nil {
+				errs = packer.MultiErrorAppend(errs, err)
+			}
+			if r.Provider != "" && !strings.EqualFold(r.Provider, "Microsoft.ManagedIdentity") {
```

Comment:
when is `r.Provider == ""` a valid option? Shouldn't the provider always be `Microsoft.ManagedIdentity`?

Reply:
Since the `err` from `client.ParseResourceID` could be non-nil value, in which case, parsed resource id `r` will hold default values for its members (empty strings for string type). So if that check fails, there's no further need to check the `Provider` and `ResourceType` names. Checking for non-empty strings prevents us from getting three different errors from a broken resource id which fails the first check and gives us more precise and targeted errors.

Reply:
Since they are just resource ids, the provider could be any valid providers. So checking for the managed identity RP here.

Reply:
fixed. was trying to avoid further checking provider and resource type strings if resource id is invalid.